frontend/src/components/common/Pagination.tsx

```diff
@@ -50,15 +50,22 @@ const Pagination = ({
       return;
     }
     setUserPageNum(newUserPageNum);
-    if (!Number.isNaN(newUserPageNum) && newUserPageNum !== pageNum) {
-      getRecords(newUserPageNum);
-    }
   };
+  
+  const fetchRecords = () => {
+    if (!Number.isNaN(userPageNum) && userPageNum !== pageNum) {
+      getRecords(userPageNum);
+    }
+  }
 
-  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
```

Comment:
So we actually want to keep this in here because we want to keep this behaviour:
- Clear the text field completely 
- Click out of the text field (triggering onBlur)
- The current page number is filled back into the text box

Right now, if you clear the box and click out of it, the box stays empty, which isn't what we want

Reply:
Ah yeah good catch, will fix - should not have written this at 12 am clearly 💀